flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/resource/ResourceManager.java

```diff
@@ -83,6 +83,56 @@ public ResourceManager(ReadableConfig config, MutableURLClassLoader userClassLoa
         this.userClassLoader = userClassLoader;
     }
 
+    public boolean exists(Path filePath) throws IOException {
+        return FileSystem.getUnguardedFileSystem(filePath.toUri()).exists(filePath);
+    }
+
+    /**
+     * Register the filePath of flink filesystem. If it is remote filesystem and the file exists
+     * then download the file at local. register the filePath map to localURL.
+     */
+    public void registerFsResources(Path filePath) throws IOException {
```

Comment:
File resource shares almost the same logic like jar resource, maybe we can extract the common utils to avoid repeatation.